packages/panzer/disc-fe/src/evaluators/Panzer_Integrator_GradBasisCrossVector_impl.hpp

```diff
@@ -137,18 +137,20 @@ namespace panzer
 
     // Create the fields that we're either contributing to or evaluating
     // (storing).
-    fields_.clear();
-    for (const auto& name : resNames)
+    fields_ = Kokkos::View<PHX::MDField<ScalarT, Cell, BASIS>*>("Integrator_GradBasisCrossVector::fields_", resNames.size());
+
     {
-      MDField<ScalarT, Cell, BASIS> res(name, basis.functional);
-      fields_.push_back(res);
+      int i=0;
+      for (const auto& name : resNames)
+        fields_(i++) = MDField<ScalarT, Cell, BASIS>(name, basis.functional);
     } // end loop over resNames
```

Comment:
I'm not understanding what this block is doing without the `for` at the top of it.  Does the whole block need to be removed?

Reply:
It's assigning the values to fields_...
I guess I don't understand your question.  It's a loop over resNames that allocs data and assigns it

Reply:
Nevermind&mdash;it was just the way the diff was rendered that confused me.  If I were you, I'd remove the surrounding `{}` and outdent lines 143&ndash;145.